pkgs/tools/misc/online-judge-tools/default.nix

```
@@ -0,0 +1,42 @@
+{ git, lib, python3Packages, ... }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "online-judge-tools";
+  version = "11.5.1";
+  format = "wheel";
+
+  src = python3Packages.fetchPypi {
+    inherit version format;
+    pname = "online_judge_tools";
+    dist = "py3";
+    python = "py3";
+    sha256 = "18avv6icax7mxnk8mbhpg1if5myadplr97pwfak5yw9hnh9z97r0";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    online-judge-api-client
+    colorama
+    requests
+  ];
+
+  # Wheels have no tests
+  doCheck = false;
+
+  preFixup = ''
+    makeWrapperArgs=(
+      --prefix PATH : ${lib.makeBinPath [ git ]}
+      --prefix PYTHONPATH : ${
+        with python3Packages;
+        python3Packages.makePythonPath [ colorlog jinja2 markdown toml ]
+      }
+    )
+  '';
```

```suggestion
    makeWrapperArgs = [
      "--prefix PATH : ${lib.makeBinPath [ git ]}"
    ];
```

The python path is already exported by `buildPythonXXX`

---

@jonringer 
These Python modules is not for online-judge-tools, but for [library-checker-problems](https://github.com/yosupo06/library-checker-problems) that online-judge-tools clones and run internally.

First I tried all these dependencies to `propagatedBuildInputs`, but it didn't work.
Is there a smarter way to solve this?

---

`propagatedBuildInputs` should work

---

@jonringer I tried again, and it didn't work.

```nix
  propagatedBuildInputs =
    [ colorama colorlog jinja2 markdown online-judge-api-client requests toml ];
```

```
$ ./result/bin/oj dl https://judge.yosupo.jp/problem/predecessor_problem
[INFO] online-judge-tools 11.5.1 (+ online-judge-api-client 10.10.0)
[INFO] load cookie from: /home/sei40kr/.local/share/online-judge-tools/cookie.jar
git version 2.33.1
[INFO] $ /nix/store/zhf7xxp7k1605wirn29kylg1x2bl4pq3-git-2.33.1/bin/git -C /home/sei40kr/.cache/online-judge-tools/library-checker-problems pull
Already up to date.
[INFO] $ /nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9 /home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py /home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml
Traceback (most recent call last):
  File "/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py", line 20, in <module>
    import toml
ModuleNotFoundError: No module named 'toml'
[ERROR] the generate.py failed: check https://github.com/yosupo06/library-checker-problems/issues
[ERROR] Command '['/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/main.py", line 101, in main
    sys.exit(run_program(parsed, parser=parser))
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/main.py", line 61, in run_program
    if not subcommand_download.run(args):
  File "/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages/onlinejudge_command/subcommand/download.py", line 120, in run
    samples = problem.download_sample_cases(session=sess)
  File "/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages/onlinejudge/service/library_checker.py", line 78, in download_sample_cases
    self._generate_test_cases_in_cloned_repository()
  File "/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages/onlinejudge/service/library_checker.py", line 108, in _generate_test_cases_in_cloned_repository
    subprocess.check_call(command, stdout=sys.stderr, stderr=sys.stderr)
  File "/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/bin/python3.9', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/generate.py', '/home/sei40kr/.cache/online-judge-tools/library-checker-problems/datastructure/predecessor_problem/info.toml']' returned non-zero exit status 1.
```

---

@jonringer 
It seems `propagatedBuildInputs` are appended to `sys.path` by `site.addsitedir`.

```python
import sys;import site;import functools;sys.argv[0] = '/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/bin/oj';functools.reduce(lambda k, p: site.addsitedir(p, k), ['/nix/store/2hwcphm0fbsia17frii22nac3z6cj1s3-online-judge-tools-11.5.1/lib/python3.9/site-packages','/nix/store/wark7kwccm3kn06h5xwhhvv5rcg8331i-python3.9-colorama-0.4.4/lib/python3.9/site-packages','/nix/store/wvncjga0w2hvsnszwa7j16p57wdkgh0y-python3.9-colorlog-6.5.0/lib/python3.9/site-packages','/nix/store/jpdalbpwrqp3gsn1cnjzivmsr390s982-python3.9-Jinja2-3.0.2/lib/python3.9/site-packages','/nix/store/z84rmnhy1grx4jls23n5aay25rp29aad-python3.9-Babel-2.9.1/lib/python3.9/site-packages','/nix/store/rj4cwhhkr8hm2w1xxw26b5kmka9541hn-python3.9-pytz-2021.3/lib/python3.9/site-packages','/nix/store/hfvwfl2pvlj5x2dm85vgfgmkll627dgd-python3.9-markupsafe-2.0.1/lib/python3.9/site-packages','/nix/store/qsqgwd1y8m47vlch7qw6qpx90zwg90ql-python3.9-Markdown-3.3.4/lib/python3.9/site-packages','/nix/store/vsz2g3a9hvbax771kicngc7yqc0pcppw-python3.9-online-judge-api-client-10.10.0/lib/python3.9/site-packages','/nix/store/0fijgd5j5g6l4gxkdlmpbs8dq1s8lg3f-python3.9-appdirs-1.4.4/lib/python3.9/site-packages','/nix/store/r3fh82iy0ialr2rivl86c6hw6ppah4gn-python3.9-beautifulsoup4-4.10.0/lib/python3.9/site-packages','/nix/store/arhdy6kri2yrb22w2an6lf0f59zhcn70-python3.9-html5lib-1.1/lib/python3.9/site-packages','/nix/store/7ixnxb28f4sg2c08a1pnp3yxdb0h48cs-python3.9-six-1.16.0/lib/python3.9/site-packages','/nix/store/bwyfa04hx5rm9k01vg13iizhs2gzdbnq-python3.9-webencodings-0.5.1/lib/python3.9/site-packages','/nix/store/2xwdhgbr58zx3b394acj4wa82bxv2v1y-python3.9-lxml-4.6.3/lib/python3.9/site-packages','/nix/store/dgnzpgfnkm78kv24r4r02ylk4lzjpbsa-python3.9-soupsieve-2.2.1/lib/python3.9/site-packages','/nix/store/jy7nn3w05g60ns5vrlw3ywjnkd5vpyx1-python3.9-jsonschema-3.2.0/lib/python3.9/site-packages','/nix/store/n8bc0fxv5sp10s9xrpr210dzj5pv5x7i-python3.9-attrs-21.2.0/lib/python3.9/site-packages','/nix/store/j192qx3qyr5w8050qq9s4p001k7d6w52-python3.9-importlib-metadata-4.8.1/lib/python3.9/site-packages','/nix/store/2vhiiavpj8d98x9ja3c80ra985wpx6g6-python3.9-toml-0.10.2/lib/python3.9/site-packages','/nix/store/kwkj2bk4zd08sd7j5qrhkxfkjk4cif45-python3.9-zipp-3.6.0/lib/python3.9/site-packages','/nix/store/lscg5l570grdka03jymahqhgcdnp0c35-python3.9-more-itertools-8.10.0/lib/python3.9/site-packages','/nix/store/b7w6b69hgcxfmhsm4hq9xwygbym0hc37-python3.9-pyrsistent-0.18.0/lib/python3.9/site-packages','/nix/store/wnd6rnxc2hqsam2r280xv4yy1bygd08n-python3.9-requests-2.26.0/lib/python3.9/site-packages','/nix/store/q6zv2g1pyjfzxphgsvr0qbm2xbfvx7jq-python3.9-certifi-2021.10.08/lib/python3.9/site-packages','/nix/store/bx6dm7hzd6dnbaifj0ydhiwxfqf90125-python3.9-idna-3.2/lib/python3.9/site-packages','/nix/store/lg6hps3px9isdhcm5jhv4x3c3xwpz3i7-python3.9-urllib3-1.26.7/lib/python3.9/site-packages','/nix/store/c8izknabvsnbrmqahmpd5pgdm0wz1pdy-python3.9-brotli-1.0.9/lib/python3.9/site-packages','/nix/store/z262iy04g0lj2g4w5gqks3i7n4dm28fc-python3.9-pysocks-1.7.1/lib/python3.9/site-packages','/nix/store/ichac696i1qr8dxvmhipnxp1af4cdfgi-python3.9-chardet-4.0.0/lib/python3.9/site-packages','/nix/store/kzb17mnvgci0mhs3pa8hdga67f8nmqxj-python3.9-brotlicffi-1.0.9.2/lib/python3.9/site-packages','/nix/store/4flpmcvw0p795bx1s9lnam7kyl7qhviv-python3.9-pycparser-2.20/lib/python3.9/site-packages','/nix/store/wf8vlq8gibp76vp816yw0fxjlc5ldphs-python3.9-cffi-1.14.6/lib/python3.9/site-packages','/nix/store/hmci2h8g3rydi7qrc9h2nmyvw9kff8l9-python3.9-charset-normalizer-2.0.5/lib/python3.9/site-packages'], site._init_pathinfo());
```

But `sys.path` is not exported to a subprocess, so I think we need another fix for this.